gravitee-rest-api-service/src/test/java/io/gravitee/rest/api/service/EnvironmentService_FindByUserTest.java

```diff
@@ -0,0 +1,125 @@
+
+import io.gravitee.repository.exceptions.TechnicalException;
+import io.gravitee.repository.management.api.EnvironmentRepository;
+import io.gravitee.repository.management.model.Environment;
+import io.gravitee.rest.api.model.EnvironmentEntity;
+import io.gravitee.rest.api.service.common.GraviteeContext;
+import io.gravitee.rest.api.service.impl.EnvironmentServiceImpl;
+import org.assertj.core.api.Assertions;
+import org.junit.After;
+import org.junit.Test;
+import org.junit.runner.RunWith;
+import org.mockito.InjectMocks;
+import org.mockito.Mock;
+import org.mockito.junit.MockitoJUnitRunner;
+
+import java.util.Arrays;
+import java.util.Collections;
+import java.util.HashSet;
+import java.util.List;
+import java.util.Set;
+
+import static org.assertj.core.api.Assertions.assertThat;
+import static org.mockito.ArgumentMatchers.any;
+import static org.mockito.Mockito.when;
+
+/**
+ * @author Yann TAVERNIER (yann.tavernier at graviteesource.com)
+ * @author GraviteeSource Team
+ */
+@RunWith(MockitoJUnitRunner.class)
+public class EnvironmentService_FindByUserTest {
+
+    @InjectMocks
+    private EnvironmentServiceImpl environmentService = new EnvironmentServiceImpl();
+
+    @Mock
+    private EnvironmentRepository mockEnvironmentRepository;
+
+    @Mock
+    private OrganizationService mockOrganizationService;
+
+    @Mock
+    private ApiHeaderService mockAPIHeaderService;
+
+    @Mock
+    private PageService mockPageService;
+
+    @After
+    public void tearDown() {
+        GraviteeContext.cleanContext();
+    }
+
+    @Test
+    public void shouldFindByUserFilteredByEnvId() throws TechnicalException {
+
+        when(mockEnvironmentRepository.findByOrganization(any())).thenReturn(getEnvironments());
+
+        List<EnvironmentEntity> environments = environmentService.findByUser(null, "fakeEnvId", null);
+
+        assertThat(environments).isEmpty();
+    }
+
+
+    @Test
+    public void shouldFindByUserFilteredByEnvHrid() throws TechnicalException {
+
+        when(mockEnvironmentRepository.findByOrganization(any())).thenReturn(getEnvironments());
+
+        List<EnvironmentEntity> environments = environmentService.findByUser(null, null, "fakeHrid");
+
+        assertThat(environments).isEmpty();
+    }
+
+    @Test
+    public void shouldFindByUserFilteredByEnvHridAndId_different() throws TechnicalException {
+
+        when(mockEnvironmentRepository.findByOrganization(any())).thenReturn(getEnvironments());
+
+        List<EnvironmentEntity> environments = environmentService.findByUser(null, "envId", "env2");
+
+        assertThat(environments).hasSize(2);
+    }
+
+    @Test
+    public void shouldFindByUserFilteredByEnvHridAndId_same() throws TechnicalException {
+
+        when(mockEnvironmentRepository.findByOrganization(any())).thenReturn(getEnvironments());
+
+        List<EnvironmentEntity> environments = environmentService.findByUser(null, "env1", "env1");
+
+        assertThat(environments).hasSize(1);
+    }
```

Comment:
These 2 cases feel weird at first read, I would have expected to get an empty list as a result because there is no env matching parameter couple (id, hrid). Do we really need to have 2 parameters? Maybe a single one `idOrHrid` would make more sense?